.circleci/config.yml

```diff
@@ -11,55 +11,109 @@ executors:
     working_directory: /tmp/storybook
     docker:
       - image: circleci/node:10-browsers
+        environment:
+          NODE_OPTIONS: --max_old_space_size=4096
     resource_class: <<parameters.class>>
 
 jobs:
-  install:
-    executor: sb_node
+  build:
+    executor:
+      class: large
+      name: sb_node
     steps:
-      - checkout
+      - add_ssh_keys:
+          fingerprints:
+            - 35:aa:b9:c3:b8:1c:27:1f:02:0a:73:1a:ac:58:77:df
+      - run:
+          name: Clone GitHub repository
+          command: echo -e "Host github.com\n\tStrictHostKeyChecking no\n" > ~/.ssh/config && git clone -b ${CIRCLE_BRANCH} --single-branch --depth=1 git@github.com:storybookjs/storybook.git /tmp/storybook
+      - run:
+          name: Remove examples
+          command: rm -rf examples/
       - restore_cache:
-          name: Restore core dependencies cache
+          name: Restore Yarn cache
           keys:
-            - core-dependencies-v5-{{ checksum "yarn.lock" }}
-            - core-dependencies-v5-
+            - build-yarn-cache-v1--{{ checksum "yarn.lock" }}
+            - build-yarn-cache-v1--
       - run:
           name: Install dependencies
-          command: yarn install
+          command: yarn install --frozen-lockfile --cache-folder ~/.cache/yarn
       - run:
-          name: Check that yarn.lock is not corrupted
-          command: yarn repo-dirty-check
+          name: Bootstrap
+          command: yarn bootstrap --core
       - save_cache:
-          name: Cache core dependencies
-          key: core-dependencies-v5-{{ checksum "yarn.lock" }}
+          name: Save Yarn cache
+          key: build-yarn-cache-v1--{{ checksum "yarn.lock" }}
           paths:
-            - node_modules
+            - ~/.cache/yarn
       - persist_to_workspace:
           root: .
           paths:
-            - node_modules
-            - examples
             - addons
             - dev-kits
             - app
             - lib
-  build:
-    executor: sb_node
+  install-examples-deps:
+    executor:
+      class: large
+      name: sb_node
     steps:
-      - checkout
-      - attach_workspace:
-          at: .
+      - add_ssh_keys:
+          fingerprints:
+            - 35:aa:b9:c3:b8:1c:27:1f:02:0a:73:1a:ac:58:77:df
       - run:
-          name: Bootstrap
-          command: yarn bootstrap --core
+          name: Clone GitHub repository
+          command: echo -e "Host github.com\n\tStrictHostKeyChecking no\n" > ~/.ssh/config && git clone -b ${CIRCLE_BRANCH} --single-branch --depth=1 git@github.com:storybookjs/storybook.git /tmp/storybook
```

Comment:
Can you explain this a bit?

Reply:
Sure! I did this after having read https://discuss.circleci.com/t/checkout-depth/18699 and https://medium.com/equify-tech/from-5-minutes-to-5-seconds-customize-your-checkout-step-in-circleci-2-0-244d7b226d7d

But sadly I just reverted this commit as this custom checkout wasn't working with forks for at least 2 reasons:
 - the URL of the repo is the one from Storybook and not the fork (I didn't find a way to get the URL of the fork)
 - `CIRCLE_BRANCH` doesn't contain a branch name in case of a PR from a fork but something like `PR/1234`